addons/binding/org.openhab.binding.http/ESH-INF/config/config.xml

```diff
@@ -0,0 +1,143 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<config-description:config-descriptions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:config-description="http://eclipse.org/smarthome/schemas/config-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/config-description/v1.0.0
+	http://eclipse.org/smarthome/schemas/config-description-1.0.0.xsd">
+
+	<config-description uri="thing-type:http:config-stateonly">
+		<parameter name="stateUrl" type="text" required="true">
+			<label>State URL</label>
+			<description>
+				URL to fetch to retrieve the state of the channel. The URL should return a State value; see the README
+				for more information.
+			</description>
+		</parameter>
+		<parameter name="stateRefreshInterval" type="integer" required="true">
+			<label>State Refresh Interval (ms)</label>
+			<description>How often (in milliseconds) to refresh state by fetching the State URL.</description>
+			<default>60000</default>
+		</parameter>
+		<parameter name="stateUsername" type="text">
+			<label>State Username</label>
+			<description>Optional username to use for HTTP Basic authentication.</description>
+		</parameter>
+		<parameter name="statePassword" type="text">
+			<context>password</context>
+			<label>State Password</label>
+			<description>Optional password to use for HTTP Basic authentication</description>
+		</parameter>
+		<parameter name="stateConnectTimeout" type="integer" required="true">
+			<label>State Connect Timeout</label>
+			<description>HTTP connect timeout, in milliseconds.</description>
+			<default>3000</default>
+		</parameter>
+		<parameter name="stateRequestTimeout" type="integer" required="true">
+			<label>State Request Timeout</label>
+			<description>HTTP request timeout, in milliseconds.</description>
+			<default>5000</default>
+		</parameter>
+		<parameter name="stateResponseTransform" type="text">
+			<label>State Response Transformation</label>
+			<description>A transformation rule used to transform the HTTP response from the State URL.</description>
+		</parameter>
+	</config-description>
+
+	<config-description uri="thing-type:http:config-all">
```

Comment:
Do you think it's not to inflexible to have a single URL for each thing?  One thing I was thinking is that a device (or whatever) might have different URLs for fetching state vs. sending commands.  Not sure how likely that is, but my feeling here is that this binding's main purpose is to deal with wacky things that don't fit into the standard openHAB way of doing things, so greater flexibility might be warranted.

Reply:
The suggestion is because of complex data types like json/xml, I guess? Why didn't we go for this Bridge/Thing/Thing architecture in MQTT? Because MQTT topic payloads can be complex data types as well, exactly like http. We still went the route of Bridge/Thing/Dynamic channels with the cost of having more configuration per channel.
If we change the architecture here, the mqtt binding would need to be adopted as well, and my coap binding currently in development would use that architecture, too. But please, we should stay consistent for all those generic protocols.

Reply:
To slightly backtrack on my last comment, I actually just realized that the example I have in the README actually becomes impossible with the proposal to move the URL config into the Thing.  (I guess you can technically create the same sitemap, but you can't have the "door switch" and "door camera" channels on the same Thing anymore.)

Would it make sense to have a sort of "master URL" on the Thing, and then allowing the channel config to override it?  That way, the common case of only needing one URL would be easy to configure.  Or is that too confusing from a UX perspective?

Reply:
The Thing could be for the base-url part ("https://myserver.com/api/") with authentication+cache config, and a channel has the relative url part only ("sensor/door") and optional transformations.

Reply:
> We'd probably require multiple relative paths (separate ones for sending & receiving) then, right?

I would make the sending URL optional and advanced, I think. Most http services are crud rest endpoints nowadays. Wdyt?